### PR TITLE
initial swag at the partner sync API spec

### PR DESCRIPTION
**Pull request by rtewalt**
Spec to inform retailers that want to integrate with the platform the webhook API's they will have to create for the integration to work.

A couple of places I left placeholders for <insert retailer info here> kind of things, as well as a placeholder for where our greater documentation would live for platform onboarding.

### CODE REVIEW

File: partner-reverse-swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      - "application/json"
+      parameters:
+      - in: "body"
+        name: "body"
+        description: "New product association object"
+        required: true
+        schema:
+          $ref: "#/definitions/ProductAssociation"
+      responses:
+        204:
```
Comment (vadimuchitel) [inline partner-reverse-swagger.yaml]: @rtewalt Why not 201 if "product association `Created`"?

Comment (atcrawford) [inline partner-reverse-swagger.yaml]: @rtewalt Why `204 No Content` instead of `201 Created`?

Comment (blackmichael) [reply partner-reverse-swagger.yaml]: I agree, `201` seems more appropriate here.

File: partner-reverse-swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      retailer_sku:
+        type: "string"
+      references:
+        type: "array"
+        items:
+          $ref: "#/definitions/References"
+  References:
+    type: "object"
+    properties:
+      retailer_location_id:
```
Comment (atcrawford) [inline partner-reverse-swagger.yaml]: @rtewalt what is the difference between a `retailer_location_id` and a `location_id`?

File: partner-reverse-swagger.yaml (hunk 1 above)
Comment (rtewalt) [reply partner-reverse-swagger.yaml]: We could.

This is a webhook to be fulfilled by our partners.  So i'm trying to keep it as simple as possible.  201 normally requires payload to be returned (which i couldn't brainstorm anything we need back) and should fill in a Location Header to retrieve the resource back.  Where as 204 basically means "yep, got the thing".

File: partner-reverse-swagger.yaml (hunk 2 above)
Comment (rtewalt) [reply partner-reverse-swagger.yaml]: location_id is the shipt id, my thought was that adding shipt_ to all of our ids would be redundant.

File: partner-reverse-swagger.yaml (hunk 1 above)
Comment (rtewalt) [reply partner-reverse-swagger.yaml]: see answer below

File: partner-reverse-swagger.yaml (hunk 2 above)
Comment (atcrawford) [reply partner-reverse-swagger.yaml]: ah, so location_id is internal and retailer_location_id is external?

Comment (rtewalt) [reply partner-reverse-swagger.yaml]: right, i was trying to make the contract shipt centric

File: partner-reverse-swagger.yaml (hunk 1 above)
Comment (rtewalt) [reply partner-reverse-swagger.yaml]: i'm open to either, although i do have reservations about the retailer side implications if we ask them to support 201

File: partner-reverse-swagger.yaml (hunk 2 above)
Comment (atcrawford) [reply partner-reverse-swagger.yaml]: cool, that sounds good to me

File: partner-reverse-swagger.yaml (hunk 1 above)
Comment (atcrawford) [reply partner-reverse-swagger.yaml]: After doing some more reading on 204, I think I do actually like that approach better. I've always interpreted that response as something that doesn't say if the action succeeded or failed, but rather just a "I got it, you can go away now" type response. However after a little reading I see that it does indeed communicate that the action succeeded, which sounds just fine to me here.

Comment (atcrawford) [reply partner-reverse-swagger.yaml]: https://httpstatuses.com/204

File: partner-reverse-swagger.yaml (hunk 2 above)
Comment (rtewalt) [reply partner-reverse-swagger.yaml]: @atcrawford i added descriptions to the fields to hopefully make it more clear

File: partner-reverse-swagger.yaml (hunk 1 above)
Comment (rtewalt) [reply partner-reverse-swagger.yaml]: @blackmichael @vadimuchitel how do you guys feel?

**Pull request merged**